lib/common/timers.ts

```diff
@@ -19,14 +19,21 @@ export function patchTimer(window: any, setName: string, cancelName: string, nam
   setName += nameSuffix;
   cancelName += nameSuffix;
 
+  const tasksByHandleId: Object = {};
+
   function scheduleTask(task: Task) {
     const data = <TimerOptions>task.data;
-    data.args[0] = task.invoke;
+    data.args[0] = function() {
+      task.invoke.apply(this, arguments);
```

Comment:
No.

Before the change, `task.invoke` was used as a timer callback. Timer callbacks are executed with some other context: `this` is a `Timeout` instance in Node.js, `window` in browers, `undefined` in strict mode.

Within this change, we use another function as a timer callback, but in the function we still call `task.invoke` with the original timer callback context. As it was called before the change.

Note, that the `invoke` method of `ZoneTask`, which is called here, also passes the context further, so it ends as the original user-provided task callback context. Changing it is not good. Consider:

```javascript
setTimeout(function() {
  console.log(this);
  // should log `window` in browsers, not a `ZoneTask` instance or anything else
});
```

This doesn’t fail any Zone.js tests, though. Seems like there is a lack of tests for the timer callbacks context.